static/js/composebox_typeahead.js

```diff
@@ -855,7 +855,15 @@ exports.content_typeahead_selected = function (item, event) {
         // Stream + topic mention typeahead; close the stream+topic mention syntax
         // with the topic and the final **.
         const start = -this.token.length;
-        beginning = beginning.slice(0, start) + item + "** ";
+        if (start !== 0) {
+            // We do not perform string slicing when the token
+            // length is 0 as it returns an empty string.
+            // The token length can be 0 for this case as we
+            // allow the setting up of the topic typeahead for
+            // completing the stream mention.
+            beginning = beginning.slice(0, start);
+        }
+        beginning = beginning + item + "** ";
```

Comment:
Could this have been a regression caused by 48f5e5179a1dab919e831909dea6867484325b1f?

```
     } else if (this.completing === "topic_list") {
         // Stream + topic mention typeahead; close the stream+topic mention syntax
         // with the topic and the final **.
-        const start = beginning.length - this.token.length;
-        beginning = beginning.substring(0, start) + item + "** ";
+        const start = -this.token.length;
+        beginning = beginning.slice(0, start) + item + "** ";
```

@andersk FYI.

Reply:
Yeah, that seems likely. We can just change `start = -this.token.length` back to `beginning.length - this.token.length` instead of adding a conditional.

Reply:
If we always just want to pass this output to `slice` and really don't like conditionals, we could also do:

```js
const start = -this.token.length || undefined;
```

Reply:
I think the original code was clearer, and at most needs a comment (especially now that we have a test that would catch our regressing this), so I think I'll try to restore that with a comment.